apis/acmpca/v1alpha1/certificateauthoritypermission_types.go

```diff
@@ -57,9 +57,10 @@ type CertificateAuthorityPermissionParameters struct {
 	// +immutable
 	Actions []string `json:"actions,omitempty"`
 
-	// The AWS Service or identity
-	// +optional
+	// The AWS service or identity that receives the permission. At this
+	// time, the only valid principal is acm.amazonaws.com.
 	// +immutable
+	// +kubebuilder:default=acm.amazonaws.com
 	Principal *string `json:"principal,omitempty"`
```

Comment:
Good idea to default this. Given we're doing that, would it hurt to make this required? Presumably anyone who had omitted it would have it defaulted when they updated (plus this is a v1alpha1 API). I'd prefer to match the fact that this is required by the upstream AWS API.

Reply:
I [tried](https://github.com/crossplane/provider-aws/pull/728#discussion_r652804560) creation with required+defaulting and it didn't work but I just tested your assumption and it seems like update does go through and adds this value to the missing ones. I'll make it required again cc @hasheddan